server/controllers/finance/journal/index.js

```diff
@@ -300,10 +305,19 @@ function editTransaction(req, res, next) {
 // converts all valid posting journal editable columns into data representations
 // returns valid errors for incorrect data
 // @TODO Many requests are made vs. getting one look up table and using that - this can be greatly optimised
-function transformColumns(rows, newRecord) {
+function transformColumns(rows, newRecord, currentTransaction) {
   const ACCOUNT_NUMBER_QUERY = 'SELECT id FROM account WHERE number = ?';
   const ENTITY_QUERY = 'SELECT uuid FROM entity_map WHERE text = ?';
   const REFERENCE_QUERY = 'SELECT uuid FROM document_map  WHERE text = ?';
+  const EXCHANGE_RATE_QUERY = `
+    SELECT ? * IF(enterprise.currency_id = ?, 1, GetExchangeRate(enterprise.id, ?, ?)) AS amount FROM enterprise
+    JOIN project ON enterprise.id = project.enterprise_id WHERE project.id = ?;
+  `;
+
+  // these are global/shared properties of the current transaction
+  const projectId = currentTransaction[0].project_id;
```

Comment:
It looks like both the client and the server logic are making use of which properties are shared on each row in a transaction - potentially it would good to define these columns in a constant definition that can be used by both the client and server. (transaction shared vs. transaction unique columns).